src/navability/entities/InferenceTypes.jl

```diff
@@ -8,7 +8,23 @@ functions inside of factors.
 """
 abstract type InferenceType end
 
-#TODO maybe generate all ZInferenceTypes with a macro
+macro nvaZInferenceType(inferencetype)
+  return esc(quote
+      Base.@__doc__ struct $inferencetype <: NVA.InferenceType 
```

Comment:
is that the same as `Base.@kwdef`?

Reply:
No, that part is covered by line 16.
This macro is just to reduce code maintenance a bit with standard Z field types.